src/AudioContext.js

```diff
@@ -0,0 +1,7 @@
+import React from "react";
```

Comment:
I feel like this is the wrong design pattern. Adding this wrapper everywhere for sound effects feels wrong to me, for a few reasons:

- This consumer pattern I think is typically used for providing state/data to the underlying components. In this case, we're not using it that way, the components don't care about the provider.

- By using this pattern, we're forced to wire up all these event handlers manually, and it also makes things cumbersome if we ever want to do something on mouse enter or mouse leave other than deal with sound effects

- It just touches every piece of code, and if you forget to add the consumer wrapper, you're going to introduce a bug

I think a better way to do this is to use composition or ES6 decorators to decorate the react components that we want to emit mouse over and mouse leave events to the scene. If the decorator case works, you'd just have to add one line to each component and it would start emitting events which could then be handled by a central event handler.

Reply:
Here's an example: https://medium.com/@gigobyte/enhancing-react-components-with-decorators-441320e8606a

So we'd add `@emitSoundEvents` or something to the top of the components we want to auto-wire onMouseEnter/onMouseLeave up to fire out-of-band events. We could use `event-target-shim` like we do in `store.js` and add a central handler, or bring A-Frame into the mix directly and just fire them directly into the scene (you'd need to determine the sanest way to provide access to the scene from the handler.)